### ISSUE TITLE
notable startup latency

### ISSUE THREAD

**Issue by thautwarm** [JuliaPy/PyCall.jl#1052]
It seems that the latency gets worse with newer Julia. Here are the startup tests on my local machine:

Julia v1.9.2:
```
julia> @time using PyCall
  0.427297 seconds (250.30 k allocations: 18.071 MiB, 1.88% compilation time)

julia> @time using PyCall
  1.162834 seconds (1.32 M allocations: 91.732 MiB, 5.27% gc time, 99.92% compilation time: 100% of which was recompilation)
```

Julia v1.7.3:
```
julia> @time using PyCall
  0.689246 seconds (1.16 M allocations: 81.998 MiB, 1.08% gc time, 42.07% compilation time)

julia> @time using PyCall
  0.630776 seconds (1.98 M allocations: 104.340 MiB, 2.96% gc time, 99.86% compilation time)
```

Julia v1.9 is quite slow at loading PyCall comparing to it used to be in v1.7. I wonder if there are any known reasons for the latency or possible approaches to address this?

**Comment by MilesCranmer** [comment]
Hey @thautwarm,
I was wondering if this is still an issue on v1.10 with https://github.com/JuliaLang/julia/pull/49525 merged?
Cheers,
Miles

**Comment by thautwarm** [comment]
It gets resolved since 1.10, but still applicable for earlier versions.

**Comment by stevengj** [comment]
Closing as a Julia issue that is now fixed.
